programs/system/src/lib.rs

```diff
@@ -30,66 +97,161 @@ pub fn entrypoint(
                 space,
                 program_id,
             } => {
-                let to = 1;
-                if !system_program::check_id(&keyed_accounts[from].account.owner) {
-                    info!("CreateAccount: invalid account[from] owner");
-                    Err(ProgramError::InvalidArgument)?;
-                }
-
-                if !keyed_accounts[to].account.userdata.is_empty()
-                    || !system_program::check_id(&keyed_accounts[to].account.owner)
-                {
-                    if space > 0 {
-                        info!(
-                            "CreateAccount: invalid argument space: {} accounts.userdata.len(): {}",
-                            space,
-                            keyed_accounts[to].account.userdata.len(),
-                        );
-                    } else {
-                        info!(
-                            "CreateAccount: invalid argument; account {} already in use",
-                            keyed_accounts[to].unsigned_key()
-                        );
-                    }
-                    Err(ProgramError::InvalidArgument)?;
-                }
-                if tokens > keyed_accounts[from].account.tokens {
-                    info!(
-                        "CreateAccount: insufficient tokens ({}, need {})",
-                        keyed_accounts[from].account.tokens, tokens
-                    );
-                    Err(ProgramError::ResultWithNegativeTokens)?;
-                }
-                keyed_accounts[from].account.tokens -= tokens;
-                keyed_accounts[to].account.tokens += tokens;
-                keyed_accounts[to].account.owner = program_id;
-                keyed_accounts[to].account.userdata = vec![0; space as usize];
-                keyed_accounts[to].account.executable = false;
+                create_system_account(keyed_accounts, tokens, space, program_id).map_err(
+                    |e| match e {
+                        SystemError::ResultWithNegativeTokens => {
+                            ProgramError::ResultWithNegativeTokens
+                        }
+                        _ => ProgramError::InvalidArgument,
```

Comment:
Since there's only two more of these,  can you list them all out?  Then someone can take a quick glance and conclude: "ProgramError sucks, I should work on #3085".